soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala

```diff
@@ -243,7 +244,14 @@ class QueryServer(val dsInfo: DSInfo, val caseSensitivity: CaseSensitivity, val
   private def query(explain: Boolean)(req: HttpRequest)(resp: HttpServletResponse): Unit =  {
     val servReq = req.servletRequest
     val datasetId = servReq.getParameter("dataset")
-    val analysisParam = servReq.getParameter("query")
+    val analysisParam = Option(servReq.getParameter("query")) match {
+      case Some(q) => q
+      case _ =>
+        using(servReq.getInputStream) { ins =>
```

Comment:
What else might be inside the inputStream? Nothing? Who decides to either put a query either in the parameter query or in the inputSteam (= body?)?

Reply:
The body only contains the query analysis.  QC is the only client and decide where to put it.